kvserver: logging improvements for TestAtomicReplicationChange

tbg opened the pull request:
See individual commits.

Closes #145946.

tbg committed:
structlogging: clean up a log line

`start` returns `nil` on graceful shutdown.

tbg committed:
kvserver: allow tests to selectively log Ready

And use in TestAtomicReplicationChange.

Touches #145946.

tbg committed:
kvserver: log reason for replica removal

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146069)
<!-- Reviewable:end -->

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

See comment re: hot ranges logging, the return err is probably unnecessary entirely.

> Reviewed 1 of 1 files at r1, 4 of 4 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146069#-:-OPVebMp9c6FQeW6bDAn:b5wccif)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @angles-n-daemons)

___
*[`pkg/server/structlogging/hot_ranges_log.go` line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146069#-OPVdjO-8Om87s5xntTc:-OPVdjO-8Om87s5xntTd:b-1jh6vv) ([raw file](https://github.com/cockroachdb/cockroach/blob/f1c9905ac223f8e8d598398f4cadb256d88f1349/pkg/server/structlogging/hot_ranges_log.go#L120)):*
> ```Go
> func (s *hotRangesLoggingScheduler) startTask(ctx context.Context, stopper *stop.Stopper) error {
> 	return stopper.RunAsyncTask(ctx, "hot-ranges-stats", func(ctx context.Context) {
> 		if err := s.start(ctx, stopper); err != nil {
> ```

It doesn't look like this code path would ever return a non-nil error:

```
func (s *hotRangesLoggingScheduler) start(ctx context.Context, stopper *stop.Stopper) error {
	for {
		ci := CheckInterval
		if s.multiTenant {
			ci *= 5
		}
		select {
		case <-stopper.ShouldQuiesce():
			return nil
		case <-ctx.Done():
			return nil
		case <-time.After(ci):
			s.maybeLogHotRanges(ctx, stopper)
		case <-TestLoopChannel:
			continue
		}
	}
}
```

Could just drop the error return entirely.


<!-- Sent from Reviewable.io -->

tbg committed:
structlogging: remove unnecessary error return

This improves on the preceding commit - an error was never returned in the
first place.

tbg commented:
Good call - done.

bors r+

> Message ID: ***@***.***
> com>
>
